language/evm/move-to-yul/src/generator.rs

```diff
@@ -727,6 +706,23 @@ impl Generator {
                     // Only dispatch callables
                     continue;
                 }
+                let extracted_sig_opt = attributes::extract_callable_signature(fun);
+                let mut sig = ctx.create_default_solidity_signature(fun);
+                if let Some(extracted_sig) = extracted_sig_opt {
+                    let parsed_sig_opt = ctx.parse_into_solidity_signature(&extracted_sig);
```

Comment:
Parsing a solidity type or signature from a string should also be a member function of `SolidityType` (perhaps get rid of the `impl Context` inside of `solidity_type` and instead pass `ctx: &Context` where you need it?